auth/user.go

```diff
@@ -397,6 +397,13 @@ func (user *userImpl) InheritedChannels() ch.TimedSet {
 		roleSince := user.RoleNames()[role.Name()]
 		channels.AddAtSequence(role.Channels(), roleSince.Sequence)
 	}
+	if channelsPerUserThreshold := user.auth.ChannelsWarningThreshold(); channelsPerUserThreshold != nil {
```

Comment:
We said we should add a `sync.Once` or something to stop this from being repeatedly logged for the same user.

```
2021-05-20T17:58:43.619+01:00 [INF] HTTP:  #013: GET /db1/_user/ben (as ADMIN)
2021-05-20T17:58:43.619+01:00 [WRN] User ID: ben channel count: 4 exceeds 3 for channels per user warning threshold -- auth.(*userImpl).InheritedChannels() at user.go:403
2021-05-20T17:58:43.619+01:00 [INF] HTTP+: #013:     --> 200   (0.6 ms)
2021-05-20T17:58:48.251+01:00 [INF] HTTP:  #014: GET /db1/_user/ben (as ADMIN)
2021-05-20T17:58:48.252+01:00 [WRN] User ID: ben channel count: 4 exceeds 3 for channels per user warning threshold -- auth.(*userImpl).InheritedChannels() at user.go:403
2021-05-20T17:58:48.252+01:00 [INF] HTTP+: #014:     --> 200   (0.5 ms)
2021-05-20T17:58:48.699+01:00 [INF] HTTP:  #015: GET /db1/_user/ben (as ADMIN)
2021-05-20T17:58:48.699+01:00 [WRN] User ID: ben channel count: 4 exceeds 3 for channels per user warning threshold -- auth.(*userImpl).InheritedChannels() at user.go:403
2021-05-20T17:58:48.699+01:00 [INF] HTTP+: #015:     --> 200   (0.5 ms)
2021-05-20T17:58:49.055+01:00 [INF] HTTP:  #016: GET /db1/_user/ben (as ADMIN)
2021-05-20T17:58:49.056+01:00 [WRN] User ID: ben channel count: 4 exceeds 3 for channels per user warning threshold -- auth.(*userImpl).InheritedChannels() at user.go:403
2021-05-20T17:58:49.056+01:00 [INF] HTTP+: #016:     --> 200   (0.6 ms)
2021-05-20T17:58:49.394+01:00 [INF] HTTP:  #017: GET /db1/_user/ben (as ADMIN)
2021-05-20T17:58:49.395+01:00 [WRN] User ID: ben channel count: 4 exceeds 3 for channels per user warning threshold -- auth.(*userImpl).InheritedChannels() at user.go:403
2021-05-20T17:58:49.395+01:00 [INF] HTTP+: #017:     --> 200   (0.5 ms)
```

Reply:
Makes sense, added changes to ensures that the check for channels per user threshold is only performed exactly once.